cmd/vulcan-retirejs/main.go

```diff
@@ -396,27 +394,29 @@ func resolveTarget(target, assetType string) (string, error) {
 		Transport: tr,
 		Timeout:   timeout,
 	}
-	if assetType == "WebAddress" {
+	switch assetType {
+	case "WebAddress":
 		resp, err := client.Get(target)
+		if err != nil {
+			return "", err
+		}
+		t := resp.Request.URL.String()
+		if resp.Request.URL.Path == "" {
+			t = fmt.Sprintf("%s/", t)
+		}
+		return t, nil
+	case "Hostname":
+		resp, err := client.Get(fmt.Sprintf("https://%s/", target))
 		if err == nil {
-			t := resp.Request.URL.String()
-			if resp.Request.URL.Path == "" {
-				t = fmt.Sprintf("%s/", t)
-			}
-			return t, nil
+			return resp.Request.URL.String(), nil
 		}
-	}
 
-	// TODO: Consider other cases.
-	resp, err := client.Get(fmt.Sprintf("https://%s/", target))
-	if err == nil {
+		resp, err = client.Get(fmt.Sprintf("http://%s/", target))
+		if err != nil {
+			return "", err
+		}
 		return resp.Request.URL.String(), nil
 	}
 
-	resp, err = client.Get(fmt.Sprintf("http://%s/", target))
-	if err != nil {
-		return "", err
-	}
-
-	return resp.Request.URL.String(), nil
+	return "", errors.New("unexpected assettype provided")
```

Comment:
I know it's a bit nitpicking but would you put this as de `default` case of the switch?

Reply:
Of course not sit. I agree it improves a bit logic/code readability. Implemented here [b0e4453](https://github.com/adevinta/vulcan-checks/pull/145/commits/b0e4453f5bdd5e50a9d22762920f66b426d28b33)